test/jdk/jdk/jfr/jvm/TestJvmCommitIntrinsicAndEA.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.jfr.jvm;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.LockSupport;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import jdk.jfr.EventType;
+import jdk.jfr.Recording;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.test.lib.Asserts;
+import jdk.test.lib.jfr.EventNames;
+import jdk.test.lib.jfr.Events;
+
+/**
+ *
+ * A successful execution is when the JTREG test executes successfully,
+ * i.e., with no exceptions or JVM asserts.
+ *
+ * In JVM debug builds, C2 will assert as part of EscapeAnalysis and dump an hs_err<pid>.log:
+ *
+ * #
+ * # A fatal error has been detected by the Java Runtime Environment:
+ * #
+ * # Internal Error (..\open\src\hotspot\share\opto\escape.cpp:4788)
+ * # assert(false) failed: EA: missing memory path
+ *
+ * Current thread JavaThread "C2 CompilerThread2"
+ * Current CompileTask:
+ * C2:27036 1966 ! 4   java.lang.VirtualThread::run (173 bytes)
+ *
+ * ConnectionGraph::split_unique_types+0x2409 (escape.cpp:4788)
+ * ConnectionGraph::compute_escape+0x11e6 (escape.cpp:397)
+ * ConnectionGraph::do_analysis+0xf2 (escape.cpp:118)
+ * Compile::Optimize+0x85d (compile.cpp:2381)
+ * ...
+ *
+ * The error is because C2's Escape Analysis does not recognize a pattern
+ * where one input of memory Phi node is MergeMem node, and another is RAW store.
+ * This pattern is created by the jdk.jfr.internal.JVM.commit() intrinsic,
+ * which is inlined because of inlining the JFR event jdk.VirtualThreadStart.
+ *
+ * As a result, EA complains about a strange memory graph.
+ */
+
+/**
+ * @test
+ * @bug 8352696
+ * @requires vm.flagless
+ * @requires vm.hasJFR & vm.debug
+ * @library /test/lib /test/jdk
+ * @run main/othervm jdk.jfr.jvm.TestJvmCommitIntrinsicAndEA
```

Comment:
Maybe use `-Xbatch`?

Reply:
What would it give?

Reply:
Disabling background compilation makes it more deterministic?

Reply:
Yes and you might need fewer runs to trigger compilation and wait for it to finish.